dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -47,6 +55,39 @@ public CommunityRestRepository() {
         System.out.println("Repository initialized by Spring");
     }
 
+    @Override
+    protected CommunityRest createAndReturn(Context context) throws AuthorizeException {
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+        ObjectMapper mapper = new ObjectMapper();
+        CommunityRest communityRest = null;
+        try {
+            ServletInputStream input = req.getInputStream();
+            communityRest = mapper.readValue(input, CommunityRest.class);
+        } catch (IOException e1) {
+            // Should the exception print the description message or log it to a file?
+            // Could this lead to information disclosure if printed?
+            throw new UnprocessableEntityException(e1.toString());
```

Comment:
I think it's OK to throw `UnprocessableEntityException()` here. It seems to be the same behavior we used in `EPersonRestRepository`: https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java#L67

It might be good to however change this to be:
`throw new UnprocessableEntityException("Error parsing the body: " + e1.toString());`

Reply:
I agree.